shell/browser/api/electron_api_app.cc

```diff
@@ -731,17 +731,30 @@ void App::AllowCertificateError(
     bool is_main_frame_request,
     bool strict_enforcement,
     base::OnceCallback<void(content::CertificateRequestResultType)> callback) {
-  auto adapted_callback = base::AdaptCallbackForRepeating(std::move(callback));
+  auto shared_callback = std::make_shared<
```

Comment:
Can `base::RefCounted` be used here instead? `std::shared_ptr` is [banned upstream](https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++-features.md#std_shared_ptr-banned).

We do have a couple of other places where shared_ptr has snuck in so I'd go along with this if there's a strong reason for it, but our default should be to follow upstream on this.